### PR TITLE
Water

### PR DESCRIPTION
**Pull request by Minywire**
## What problem does this pull request address?

One of the bonus/core requirements for the engine was to have water within our scene. 
## How does this pull request solve the problem?

This PR solves the problem by loading in a plane model to act as the ocean/lakes within our scene and animating it using trigonometric functions to emulate the tide. 

In testing I found that doomguy was segfaulting. To fix this we added a return value for the MD2Model.cpp LoadModel function, and removed the const ref returns from BBMD2.

## What testing has been performed?

- [x] Builds successfully
- [x] Runs successfully

https://github.com/Minywire/Bottlebrush_Engine/assets/51070144/c3775e3f-76fb-478a-b644-3394a399aa5f

### CODE REVIEW

File: Bottlebrush_Engine/Bottlebrush_Engine/BBScene/Scene.cpp [hunk 1]
```diff
@@ -124,10 +125,13 @@ Scene::Scene(const std::string& lua_master, float screenwidth, float screenheigh
     setRendererShaderSource(skyboxShaderType,
                         "Resources/Shaders/Vertex/Skybox.vert",
                         "Resources/Shaders/Fragment/Skybox.frag");
-
```
Comment (jdilanzo) [inline Bottlebrush_Engine/Bottlebrush_Engine/BBScene/Scene.cpp:127]: Delete

File: Bottlebrush_Engine/Bottlebrush_Engine/BBScene/Scene.cpp [hunk 2]
```diff
@@ -166,7 +172,8 @@ void Scene::init()
     }
 
     glEnable(GL_DEPTH_TEST);
-
```
Comment (jdilanzo) [inline Bottlebrush_Engine/Bottlebrush_Engine/BBScene/Scene.cpp:169]: Delete

File: Bottlebrush_Engine/Bottlebrush_Engine/BBScene/Scene.cpp [hunk 3]
```diff
@@ -252,13 +259,13 @@ void Scene::update()
         // MY SCENE
         setProjectionMatrix(projection);
         setViewMatrix(view);
-
```
Comment (jdilanzo) [inline Bottlebrush_Engine/Bottlebrush_Engine/BBScene/Scene.cpp:255]: Delete

File: Bottlebrush_Engine/Bottlebrush_Engine/BBScene/Scene.h [hunk 4]
```diff
@@ -29,7 +30,6 @@ class Scene
      * @param lua_master
      */
     explicit Scene(const std::string & lua_master, float screenwidth, float screenheight);
-
```
Comment (jdilanzo) [inline Bottlebrush_Engine/Bottlebrush_Engine/BBScene/Scene.h:32]: Delete

File: Bottlebrush_Engine/Bottlebrush_Engine/BBScene/Scene.h [hunk 5]
```diff
@@ -145,7 +145,7 @@ class Scene
     const float UpdateAIInterval; // interval at which to update AI, a value of 1 roughly represents a second
     
     Skybox skybox;
-
```
Comment (jdilanzo) [inline Bottlebrush_Engine/Bottlebrush_Engine/BBScene/Scene.h:148]: Delete

File: Bottlebrush_Engine/Bottlebrush_Engine/EntryPoint.cpp [hunk 6]
```diff
@@ -18,6 +18,6 @@ int main() {
 
     gameScene.update();
 
+
   return 0;
 }
-
```
Comment (jdilanzo) [inline Bottlebrush_Engine/Bottlebrush_Engine/EntryPoint.cpp:23]: ???

File: Bottlebrush_Engine/Bottlebrush_Engine/Resources/Models/h2o-just-add-water.png
Comment (jdilanzo) [inline Bottlebrush_Engine/Bottlebrush_Engine/Resources/Models/h2o-just-add-water.png:1]: Same as above pls delete

File: Bottlebrush_Engine/Bottlebrush_Engine/Resources/Models/water.png
Comment (jdilanzo) [inline Bottlebrush_Engine/Bottlebrush_Engine/Resources/Models/water.png:1]: ?????

**Review by jdilanzo — COMMENTED** [review]
lgtm

**Review by jdilanzo — DISMISSED** [review]
wotor (whitespace is cringe)

**Review by jdilanzo — DISMISSED** [review]
Wotor, also need to merge main and resolve conflicts (if any)

**Review by The-Spaghetti-Is-Ready — COMMENTED** [review]
pls delete water.lua

File: Bottlebrush_Engine/Bottlebrush_Engine/Resources/Shaders/Vertex/Water.vert [hunk 7]
```diff
@@ -0,0 +1,24 @@
+#version 330 core
+//const float pi = 3.1459;
```
Comment (jdilanzo) [inline Bottlebrush_Engine/Bottlebrush_Engine/Resources/Shaders/Vertex/Water.vert:2]: Do we need?

**Comment by Minywire** [comment]
> delete

https://tenor.com/en-GB/view/delete-cybermen-dw-doctorwho-exterminate-gif-4902419

**Review by jdilanzo — DISMISSED** [review]
**W O T O R**

**Review by jdilanzo — APPROVED** [review]
Please fucking merge

**Review by The-Spaghetti-Is-Ready — APPROVED** [review]
fuck you water

**Pull request merged**